cpp/include/rmm/device_scalar.hpp

```diff
@@ -161,7 +201,16 @@ class device_scalar {
    */
   [[nodiscard]] value_type value(cuda_stream_view stream) const
   {
-    return _storage.front_element(stream);
+    if (_host_bounce_buffer) {
+      // Case: Copying with pinned host memory
+      RMM_CUDA_TRY(cudaMemcpyAsync(
+        *_host_bounce_buffer, data(), sizeof(T), cudaMemcpyDefault, stream.value()));
+      stream.synchronize();
+      return **_host_bounce_buffer;
+    } else {
+      // Case: Copying with pageable host memory — may trigger an implicit synchronization.
+      return _storage.front_element(stream);
```

Comment:
This makes me wonder if the bounce buffer copying support should be in device_buffer instead

Reply:
Hi @harrism, thanks! Yes, that’s also my main concern. It’s why I didn’t prepare a well-considered pull request but only a draft here, as there’s a chance it’ll need to be rewritten again, just like I already did in cuDF.

We can discuss where the buffer should live. cuDF places the buffer in [its scalar header](https://github.com/rapidsai/cudf/blob/branch-25.08/cpp/include/cudf/detail/device_scalar.hpp#L91), and that’s the behavior I’d like to mimic. 
Alternatively, storing it in the RMM devicevector header is also possible, since [the element call](https://github.com/rapidsai/rmm/blob/branch-25.08/cpp/include/rmm/device_uvector.hpp#L325-L334) results in a copy as well.